python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1775 @@
+            )
+            fy_list = (
+                self.tarts.fy.cpu().numpy().tolist()
+                if hasattr(self.tarts.fy, 'cpu')
+                else list(self.tarts.fy)
+            )
+
+            # Flatten nested lists if needed
+            if fx_list and isinstance(fx_list[0], list):
+                fx_list = [item for sublist in fx_list for item in sublist]
+            if fy_list and isinstance(fy_list[0], list):
+                fy_list = [item for sublist in fy_list for item in sublist]
+
+            # Use fx/fy lists if they match the number of donuts
+            if len(fx_list) == num_donuts and len(fy_list) == num_donuts:
+                donut_data["fx"] = fx_list
+                donut_data["fy"] = fy_list
+            else:
+                # If lengths don't match, pad with zeros
+                donut_data["fx"] = [fx_list[0] if len(fx_list) > 0 else 0.0] * num_donuts
+                donut_data["fy"] = [fy_list[0] if len(fy_list) > 0 else 0.0] * num_donuts
+        else:
+            donut_data["fx"] = [0.0] * num_donuts
+            donut_data["fy"] = [0.0] * num_donuts
+
+        # Get SNR from TARTS
+        if hasattr(self.tarts, 'SNR'):
+            snr_list = (
+                self.tarts.SNR.cpu().numpy().tolist()
+                if hasattr(self.tarts.SNR, 'cpu')
+                else list(self.tarts.SNR)
+            )
+
+            # Flatten nested lists if needed
+            if snr_list and isinstance(snr_list[0], list):
+                snr_list = [item for sublist in snr_list for item in sublist]
+
+            if len(snr_list) == num_donuts:
+                donut_data["snr"] = snr_list
+            else:
+                donut_data["snr"] = [snr_list[0] if len(snr_list) > 0 else np.nan] * num_donuts
+        else:
+            donut_data["snr"] = [np.nan] * num_donuts
+
+        # Add detector information as a column (required by downstream
+        # aggregation)
+        # Ensure detector name is consistent with expected format
+        donut_data["detector"] = [detector_name] * num_donuts
+        self.log.info("Using detector name: '%s' for %d donuts (defocal type: '%s')",
+                     detector_name, num_donuts, defocalType)
+
+        # Add missing telescope coordinate columns required by downstream
+        # aggregation
+        # These are not generated by neural network, so use default values
+        donut_data["thx_CCS"] = [np.nan] * num_donuts  # Telescope X in Camera Coordinate System
+        donut_data["thy_CCS"] = [np.nan] * num_donuts  # Telescope Y in Camera Coordinate System
+        donut_data["thx_OCS"] = [np.nan] * num_donuts  # Telescope X in Observatory Coordinate System
+        donut_data["thy_OCS"] = [np.nan] * num_donuts  # Telescope Y in Observatory Coordinate System
+        donut_data["th_N"] = [np.nan] * num_donuts      # Telescope North coordinate
+        donut_data["th_W"] = [np.nan] * num_donuts      # Telescope West coordinate
```

Comment:
Does downstream replace the `nan` values with defaults? Or do you need to actually provide the defaults here?